internal/contour/cluster.go

```diff
@@ -123,7 +132,7 @@ func (cc *ClusterCache) recomputeService(oldsvc, newsvc *v1.Service) {
 	}
 }
 
-func edscluster(svc *v1.Service, portString string, config *v2.Cluster_EdsClusterConfig) *v2.Cluster {
+func edscluster(svc *v1.Service, portString string, upstreamProtocols map[string]bool, config *v2.Cluster_EdsClusterConfig) *v2.Cluster {
```

Comment:
I don't think this is right. I think what is needed is a a map of portStrings to protocol. I don't think you need to pass a map into edscluster, just pass in something representing the upstream l7 protocol; ie

edscluster(svc, "admin", "h2", config{...})

It doesn't need to be a string, (maybe an enumeration like)
const (
    protocolUnknown = 0
    protocolHTTP1 = 1
    protocolHTTP2 = 2
    protocolHTTPS = 3
)

but I don't think you need to pass in a map of values if edscluster doesn't need to consult each of those values.

Reply:
> I don't think this is right. I think what is needed is a a map of portStrings to protocol. I don't think you need to pass a map into edscluster, just pass in something representing the upstream l7 protocol; ie

I agree, but I went with string I'll let you decide in further PRs how do you prefer to proceed.

> It doesn't need to be a string, (maybe an enumeration like)
> const (
> protocolUnknown = 0
> protocolHTTP1 = 1
> protocolHTTP2 = 2
> protocolHTTPS = 3
> )
> 
> but I don't think you need to pass in a map of values if edscluster doesn't need to consult each of those values.

This means that Envoy doesn't support h2c?